### ISSUE TITLE
[feature request] wavebox built in browser - (auto) hide address bar when viewing files

### ISSUE THREAD

**Issue by nmat** [wavebox/waveboxapp#442]
* Which version of Wavebox are you using?
3.4.3 beta

* Which Operating System are you using?
Windows 10

* Description:
When viewing a file in a different window. That is:

1. Go to google drive
2. Select a file for preview
3. Select "Open in new window"

The addressbar takes a lot of focus from viewing the document itself. That the address bar i available is good but auto hiding it unless the user wants to know what the address is would be better. It will also bring more focus to the document. Or am I missing a setting in wavebox that would allow me to do this?

**Comment by Thomas101** [comment]
Definitely. It's still a bit clunky from a UI perspective at the moment. I think there was some talk about adding tab support too, but I'm not too sure whats happening with that at the moment

**Comment by nmat** [comment]
@Thomas101 sounds awesome! it would help when reading at least pdfs or mails that is opened in its own window to set focus on the content rather then the addressbar. :)

**Comment by nmat** [comment]
IMO hiding the addressbar should be set to default and then use right click menu to have a   "Show addressbar".....and a preference saying that it can be on by default if you wish.

If I am not wrong. The built in browser window is a bit of a way to work with sites in a more dynamic way so addressbar might not be interesting unless asked for it?

Please correct me if I am wrong. :)

**Comment by Thomas101** [comment]
I think if we're just showing a pdf and there's no back/forward etc then the address bar only serves the purpose of adding the "open in browser" button. We can look to moving that into the pdf renderer and then removing the bar altogether in that scenario. I'll take a look to see what's possible

**Comment by nmat** [comment]
I wonder if pdf is the only scenario though?

I mean, lets say that you would want to work on a gdoc in a different window. Will that addressbar be overused in that case? I mean....doesn't it make sense to have it hidden by default to concentrate on the actual content?

Just thinking that if open in new window would actually need the addressbar unless asked for in general?

Another example can be the call for slack. It will open a addressbar also...however you do not really "need" it and it might bring the content out of focus?

**Comment by nmat** [comment]
I can however understand that using grammarly might need the bar. But then could the extension trigger the addressbar on lets say "new tab"? or by function?

**Comment by nmat** [comment]
Oh I just found another bad place this would show. I did a presentation in google slides and did the play slides now using Ctrl+F5 and the addressbar is there also :/

This was not very pretty on presentations. :/

**Comment by Thomas101** [comment]
Awesome to hear that worked :-). I haven't seen any fullscreen events in electron, we may need to add support for that in to detect when the window expands

**Comment by jason0x43** [comment]
I would like to second this request. At least for my usage, I never need/want to see the address bar in Wavebox browser windows. It would be nice if that could hidden via an option, and possibly enabled from a context menu (or an 'open in default browser' option could be in a context menu).

**Comment by nmat** [comment]
To give more feedback here:

Preferences in wavebox could have: User Interface -> Navigation Bar -> Show Navigation bar (default) -> On/Off

Per service Under Advanced -> Show Navigation bar (override default setting) -> On/Off

Context menu: Show Navigation Bar (Keybinding info)

This way you can cover all the areas. But its my 2 cents

**Comment by jason0x43** [comment]
I'm so looking forward to this. I hope it does arrive in 4.6... :)

**Comment by jason0x43** [comment]
Hmmm...interestingly, Wavebox will actually open some links without adding the address bar when they're shift-clicked. It's not _quite_ the goal, but it shows that a basic mechanism for hiding or showing the full address bar is in there somewhere.

Shift-clicked GitHub issue link:

<img width="815" alt="Screen Shot 2019-09-03 at 2 39 58 PM" src="https://user-images.githubusercontent.com/1042866/64199663-cf303e00-ce58-11e9-9607-05398fb15d8f.png">

The same link via a normal click:

<img width="820" alt="Screen Shot 2019-09-03 at 2 39 33 PM" src="https://user-images.githubusercontent.com/1042866/64199681-d9523c80-ce58-11e9-853e-1769cc15e466.png">
